bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -0,0 +1,191 @@
+
+    public LongPolling(ScheduledExecutorService scheduler, Consumer<LongPollResult> handleResult,
+            Consumer<Throwable> handleFailure) {
+        this.scheduler = scheduler;
+        this.handleResult = handleResult;
+        this.handleFailure = handleFailure;
+    }
+
+    public void start(BoschHttpClient httpClient) throws LongPollingFailedException {
+        // Subscribe to state updates.
+        String subscriptionId = this.subscribe(httpClient);
+        this.executeLongPoll(httpClient, subscriptionId);
+    }
+
+    public void stop() {
+        // Abort long polling.
+        this.aborted = true;
+        Request request = this.request;
+        if (request != null) {
+            request.abort(new AbortLongPolling());
+            this.request = null;
+        }
+    }
+
+    /**
+     * Subscribe to events and store the subscription ID needed for long polling.
+     * 
+     * @param httpClient Http client to use for sending subscription request
+     * @return Subscription id
+     */
+    private String subscribe(BoschHttpClient httpClient) throws LongPollingFailedException {
+        try {
+            String url = httpClient.createUrl("remote/json-rpc");
+            JsonRpcRequest request = new JsonRpcRequest("2.0", "RE/subscribe",
+                    new String[] { "com/bosch/sh/remote/*", null });
+            logger.debug("Subscribe: Sending request: {} - using httpClient {}", gson.toJson(request), httpClient);
+            Request httpRequest = httpClient.createRequest(url, POST, request);
+            SubscribeResult response = httpClient.sendRequest(httpRequest, SubscribeResult.class);
+
+            logger.debug("Subscribe: Got subscription ID: {} {}", response.getResult(), response.getJsonrpc());
+            String subscriptionId = response.getResult();
+            return subscriptionId;
+        } catch (TimeoutException | ExecutionException | InterruptedException e) {
+            throw new LongPollingFailedException("Error on subscribe request", e);
+        }
+    }
+
+    private void executeLongPoll(BoschHttpClient httpClient, String subscriptionId) {
+        scheduler.execute(() -> this.longPoll(httpClient, subscriptionId));
+    }
+
+    /**
+     * Start long polling the home controller. Once a long poll resolves, a new one is started.
+     */
+    private void longPoll(BoschHttpClient httpClient, String subscriptionId) {
+        logger.debug("Sending long poll request");
+
+        JsonRpcRequest requestContent = new JsonRpcRequest("2.0", "RE/longPoll", new String[] { subscriptionId, "20" });
+        String url = httpClient.createUrl("remote/json-rpc");
+        Request request = httpClient.createRequest(url, POST, requestContent);
+
+        // Long polling responds after 20 seconds with an empty response if no update has happened
+        request.timeout(30, TimeUnit.SECONDS);
+
+        this.request = request;
+        try {
+            ContentResponse contentResponse = request.send();
```

Comment:
@fwolter Thanks for the report. The link to the documentation is broken. But I already used Jetty asynchronous, so I changed the long polling requests to be async instead of synchronous to not block a thread.